src/provider/tss.rs

```diff
@@ -0,0 +1,1372 @@
+                    .load(parent_key_handle.into(), private, public)
+                    .map_err(|tpm_err| {
+                        error!(?tpm_err);
+                        TpmError::TssKeyLoad
+                    })
+            },
+        )?;
+
+        // The object is now loaded, and the parent key has been released. Now we can
+        // save the context for the child key so that we don't fill up objectMemory in
+        // the context.
+        self.execute_with_temporary_object(key_handle.into(), |hsm_ctx, key_handle| {
+            hsm_ctx.tpm_ctx.context_save(key_handle).map_err(|tpm_err| {
+                error!(?tpm_err);
+                TpmError::TssContextSave
+            })
+        })
+    }
+}
+
+impl Tpm for TssTpm {
+    // create a root-storage-key
+    fn root_storage_key_create(
+        &mut self,
+        auth_value: &AuthValue,
+    ) -> Result<LoadableStorageKey, TpmError> {
+        // Setup the primary key.
+        let primary = self.setup_owner_primary()?;
+
+        let tpm_auth_value = match auth_value {
+            AuthValue::Key256Bit { auth_key } => Auth::from_bytes(auth_key.as_ref()),
+        }
+        .map_err(|tpm_err| {
+            error!(?tpm_err);
+            TpmError::TssAuthValueInvalid
+        })?;
+
+        self.execute_with_temporary_object(
+            primary.key_handle.into(),
+            |hsm_ctx, primary_key_handle| {
+                let (private, public) =
+                    hsm_ctx.create_storage_key(Some(tpm_auth_value.clone()), primary_key_handle)?;
+
+                // Now do a temporary load and create for the storage key.
+                let key_handle = hsm_ctx
+                    .tpm_ctx
+                    .load(primary_key_handle.into(), private.clone(), public.clone())
+                    .map_err(|tpm_err| {
+                        error!(?tpm_err);
+                        TpmError::TssStorageKeyLoad
+                    })?;
+
+                // Now it's loaded, create the machine storage key
+                let (sk_private, sk_public) = hsm_ctx.execute_with_temporary_object(
+                    key_handle.into(),
+                    |hsm_ctx, parent_key_handle| {
+                        hsm_ctx
+                            .tpm_ctx
+                            .tr_set_auth(parent_key_handle, tpm_auth_value)
+                            .map_err(|tpm_err| {
+                                error!(?tpm_err);
+                                TpmError::TssStorageKeyLoad
+                            })?;
+
+                        hsm_ctx.create_storage_key(None, parent_key_handle)
+                    },
+                )?;
+
+                Ok(LoadableStorageKey::TpmAes128CfbV1 {
+                    private: Some(private),
+                    public: Some(public),
+                    sk_private,
+                    sk_public,
+                })
+            },
+        )
+
+        // Remember this isn't loaded and can't be used yet!
```

Comment:
Sounds like this should be an error or something?

Reply:
Not an error, just saying that the object needs loading before it can be used. Ive removed it.